### ISSUE TITLE
[REC] Not tracking focused trickery count properly

### ISSUE THREAD

**Issue by Williamn88** [Hekili/hekili#1472]
### Before You Begin

- [x] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

For my MM hunter with 4 piece leggo I expect it to recommend casting a arcane shot/multi shot when the 4p focus count is at 0.
Instead it recommends a none buffed aimed shot.

I opened the APL conditions and looked at the focused_trickery_count and could see it was out of sync. At the moment the snap shot was generated it said it was at 25 when infact it is at 0. I can confirm this by casting a aimed shot and see it does not trick shot to other targets.

Following looking at the value it seems like it does not count focus spend on **lock n load procs** and **Vigil leggo procs** (which while in game says causes your next aimed shot to not consume focus does count towards the 4p focus spend counter) which contributes to it getting out of sync.

Image of the addon count thinking its 25 and the WA which is counting it correct is at 0 https://imgur.com/GFh7zxV (also confirmed that the WA was correct by checking the next cast if it cleaved).

Another thing I noticed, you often do bursting shots before a pull if you got time. The addon does not not count focus spend on this focused_trickery_count.

Hope I got everything.

### How to Reproduce

Be MM Hunter with 4p and vigil leggo
Watch the focused_trickery_count while spending vigil procs or LnL or do a pre pull bursting shot

### Snapshot (Link)

https://pastebin.com/kVqrt7Fi

**Comment by Williamn88** [comment]
Okay kinda ignore the pre fight bursting shot part. Blizz just fixed that so focus spend is reset  on pull :)

EDIT: On second thought maybe not ignore it as it still have its use in other content then raids.

 I did some more testing on a target dummy as I wanted to see if it counted bursting shot while in combat. That worked fine but I discovered something else. It does not seem to count the very first cast that starts combat when freshly reloaded, but only seems to happen if that cast is arcane shot.
If I start with a multi shot or aimed shot it seems fine.

Like 
1. I unequip, re-equip tier set and reloadui to reset everything
2. I cast arcane shot into a target dummy
3. I look at focused_trickery_count and its still on 0
4. Seems to only happen with arcane shot. If I drop combat and start combat again with arcane shot it does count it, it only happens on a fresh reloadui and the very first cast.
BONUS info, it seems like its the same as when I thought a pre-pull bursting shot did not count. If im freshly reloaded and do a bursting shot it does not count it, if I do another one after it does count it (without entering combat or anything)

So to sum it up. From my testing it seems like there is the following bugs.
- Vigil proc does not count towards focus count
-  Lock and load procs does not count towards count
- The very first cast when logging in/reloading ui does not count for arcane shot and bursting shot

Hope it makes sense.

**Comment by Williamn88** [comment]
Added more info above after doing some more testing.

**Comment by Hekili** [comment]
> I look at focused_trickery_count and its still on 0

Where are you looking at it, precisely?

> Vigil proc does not count towards focus count
> Lock and load procs does not count towards count

I'm looking at both of these, just deciding on how I want to solve it.

> The very first cast when logging in/reloading ui does not count for arcane shot and bursting shot

I'd like to know how you're testing this *precisely*.  I.e., if you're pausing and looking at the display, firing an arcane shot, and then checking the same paused display, the number is not going to update.

**Comment by Williamn88** [comment]
Hi Hekili

Im going into the apl, selecting any skill that got focused_trickery_count and hovering over it. Like here https://imgur.com/a/Mb73jPj 

The addon is not paused or anything.
Can try to precisely describe what im doing.

1. Going to a target dummy
2. unequip, re-equip tier set and reloadui to reset everything
3. Open Hekili and go to the priorities and select something like aimed shot from the ST action list and hover over the conditions and see  focused_trickery_count is at 0
4.  Spend 20 focus on doing a arcane shot into the dummy, see  focused_trickery_count  is still at 0
5.  Fire another arcane shot after without doing anything else and  focused_trickery_count  is now at 20 when in fact it should be at 0 due to hitting the 40 focus count for tier set (can confirm that by seeing the trick shot proc from tier)

Does it make sense or should I try to do a video showing it ?

**Comment by Hekili** [comment]
There's no need for a video, the explanation is fine.  I can test on PTR directly; I just wanted to be confident you were looking in the right places because some things are "frozen" and some things are immediately updating, so it's easy to mislead one's self.  It looks like you're largely correct.

**Comment by Williamn88** [comment]
Ah perfect. Not sure if I was able to put it into words correctly. Handy with the PTR for stuff like this.
Let me know if there is anything more I can assist with.

**Comment by Williamn88** [comment]
No rush at all, just curious if you found a smart way to tackle it ?

**Comment by Hekili** [comment]
Still working on and strategizing.  I couldn't get on PTR to test.

**Comment by pozzer18** [comment]
any update on this?

**Comment by Hekili** [comment]
I'll post when there's an update on this.

**Comment by Hekili** [comment]
Closing; set bonus will be disabled soon.
